csrc/scheduler/tools/inlining.cpp

```diff
@@ -193,6 +193,36 @@ size_t MaxPosCalculator::getMaxProducerPosFromConsumer(
     }
     return producer->nDims();
   } else {
+    // Gather sets of loop broadcasts (broadcast domains that are not connected
+    // to the logical domain)
+    std::unordered_set<Val*> all_additional_ids{
+        producer->domain()->additionalIDs().begin(),
+        producer->domain()->additionalIDs().end()};
+    all_additional_ids.insert(
+        consumer->domain()->additionalIDs().begin(),
+        consumer->domain()->additionalIDs().end());
+    std::unordered_set<Val*> loop_broadcasts;
+    for (TensorView* tv : {producer, consumer}) {
+      for ([[maybe_unused]] auto [expr, dir] : IRBFS::getExprsBetween(
+               {tv->domain()->additionalIDs().begin(),
+                tv->domain()->additionalIDs().end()},
+               {tv->getLoopDomain().begin(), tv->getLoopDomain().end()},
+               /*require_all_to_visited=*/false)) {
```

Comment:
This includes all IDs that are between `additionalIDs()` and loop domain. However, we could have something like this:
```
tv->broadcast(0, 16);
tv->merge(0);
```
In this case, we'll be merging the new broadcast ID with a pre-existing loop ID, so we should not ignore that. I think instead maybe what we should do is traverse from the root domain to the loop domain instead and the complement will then be the "pure" loop broadcasts which we can ignore.

Reply:
Actually I suppose that would also automatically allow us to inline past regular broadcasts that are created using BroadcastOp, but we already inline past those IDs anyway I believe.